saidump/saidump.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+extern "C" {
+#include <sai.h>
+}
+#include <inttypes.h>
+#include <set>
+#include <regex>
+#include <climits>
+#include <getopt.h>
+#include "swss/table.h"
+#include "meta/sai_serialize.h"
+#include "sairediscommon.h"
+#include <nlohmann/json.hpp>
+
+// Default value: 100MB
+static constexpr int64_t RDB_JSON_MAX_SIZE = 1024 * 1024 * 100;
+
+namespace syncd
+{
+    class SaiDump
+    {
+        public:
+            SaiDump() = default;
+            ~SaiDump() = default;
+            sai_status_t handleCmdLine(int argc, char **argv);
+            void dumpFromRedisDb();
+            void printUsage();
+            sai_status_t dumpFromRedisRdbJson();
+            sai_status_t preProcessFile(const std::string file_name);
+            void traverseJson(const nlohmann::json & jsn);
+
+        public:
+            std::string rdbJsonFile;
+            uint64_t rdbJSonSizeLimit;
+
+        private:
+            bool skipAttributes;
+            bool dumpTempView;
+            bool dumpGraph;
+            std::map<sai_object_id_t, const swss::TableMap*> g_oid_map;
```

Comment:
remove g_ prefix since this is mememer, please follow convention like in other classes, all members should be prefixed m_XXX

Reply:
Fixed.